Cuckoo hash build fails due to unspecified type conversion

Build log output:
```
/home/runner/work/highway/highway/hwy/contrib/hash/cuckoo2x2.cc: In member function ‘hwy::Cuckoo2x2Data hwy::N_PPC9::{anonymous}::PerWorkerBuilder::Take(size_t, size_t)’:
/home/runner/work/highway/highway/hwy/contrib/hash/cuckoo2x2.cc:154:46: error: conversion to ‘uint32_t’ {aka ‘unsigned int’} from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  154 |       num_primary += (choice_[i] == 0) ? 1 : 0;
      |                                              ^
```

--- a/hwy/contrib/hash/cuckoo2x2.cc
+++ b/hwy/contrib/hash/cuckoo2x2.cc
@@ -151,7 +151,7 @@ class PerWorkerBuilder {
     const size_t num_keys = choice_.size();
     uint32_t num_primary = 0;
     for (size_t i = 0; i < num_keys; ++i) {
-      num_primary += (choice_[i] == 0) ? 1 : 0;
+      num_primary += static_cast<uint32_t>((choice_[i] == 0) ? 1 : 0);
     }
     data.num_primary = num_primary;
     data.num_secondary = static_cast<uint32_t>(num_keys) - num_primary;

--- a/hwy/contrib/hash/hash_test.cc
+++ b/hwy/contrib/hash/hash_test.cc
@@ -381,7 +381,7 @@ static HWY_NOINLINE void TestMaskedBuckets(const Hash& hash) {
       const T h = hash(static_cast<T>(i));
       HWY_ASSERT_M(h <= kMask, "Output exceeds mask");
       const size_t byte_idx = static_cast<size_t>(h / 8);
-      const uint8_t bit_mask = uint8_t{1} << (h % 8);
+      const uint8_t bit_mask = static_cast<uint8_t>(uint8_t{1} << (h % 8));
       if (seen[byte_idx] & bit_mask) {
         HWY_ABORT("%s<%zu>: collision at input %zu, output %zu", Hash::Name(),
                   kBits, static_cast<size_t>(i), static_cast<size_t>(h));

